plugins/SEO-Tool/js/main.js

```diff
@@ -0,0 +1,191 @@
+		clear();
+		reports = {};
+		luColors = {};
+		colorClear(pages);
+		colorClear(LUs);
+		storage.clear();
+		document.getElementById('loader').style.display = 'block';
+		reportGen();
+	};
+
+	function getDate(ts)
+	{
+		const date = new Date(ts), days = ["Sun", "Mon", "Tue", "Wed", "Thurs", "Fri", "Sat"];
+		let dateStrg = days[date.getDay()] + ", " + date.getDate() + "/" + ('0' + String(date.getMonth())).slice(-2) + "/" + date.getFullYear() + " " + ('0' + String(date.getHours())).slice(-2) + ":" + ('0' + String(date.getMinutes())).slice(-2) + ":" + ('0' + String(date.getSeconds())).slice(-2);
+		return dateStrg;
+	};
+
+	function expiryCheck(storage) {
+		let timeStamp = JSON.parse(storage.getItem('timeStamp')), duration = JSON.parse(storage.getItem('duration'));
+		if (timeStamp === null) {     
+			timeStamp = Date.now();
+			duration = 2 * 60 * 60 * 1000;
+			storage.setItem('timeStamp', JSON.stringify(timeStamp));
+			storage.setItem('duration', JSON.stringify(duration));
+		}
+
+		else if (Date.now() > timeStamp + duration) {
+			newReport();
+			return true;
+		}
+
+		document.getElementById("timeStamp").innerHTML = getDate(timeStamp);
+		return false;
+	};
+
+	function isElement(element) {
+		return element instanceof Element || element instanceof HTMLDocument;
+	};
+
+	async function changeActive(elem) {
+		const siblingTabs = elem.parentNode.children, subtabs = document.getElementById(elem.getAttribute('data-url') + 'SubTabs');
+		Object.keys(siblingTabs).forEach((key, i) => {
+			siblingTabs[key].classList.remove('is-active');
+		});
+
+		elem.classList.add('is-active');
+		if(isElement(active) && !active.contains(elem))
+		{
+			active.classList.add('no-show');
+			active.style.display = 'none';
+			active = {};
+		}
+
+		if(subtabs === null)
+		{
+			if(!(elem.getAttribute('data-url') in reports))
+			{
+				document.getElementById('loader').style.display = 'block';
+				clear();
+			}
+
+			else
+			{
+				document.getElementById('loader').style.display = 'none';
+				populate(elem.getAttribute('data-url'), reports[elem.getAttribute('data-url')]);
+			}
+		}
+
+		else
+		{
+			subtabs.classList.remove('no-show');
+			subtabs.style.display = 'block';
+			active = subtabs;
+
+			const currTabs = document.getElementsByClassName('is-active');
+			Object.keys(currTabs).forEach((key, ind) => {
+				if(subtabs.contains(currTabs[key]))
+				{
+					changeActive(currTabs[key]);
+				}
+			});
+		}
+	};
+
+	function populate(link, report) {
+		lighthousePopulate(link, report['lighthouse']);
+		gscPopulate(link, report['gsc']);
+	};
+
+	const storage = window.localStorage, tabs = document.getElementsByClassName('v-tabs'), colors = ['red', 'orange', 'green'];
+	let active = {}, luColors = {};
+	const [pages, LUs] = parse(tabs);
+
+	const subArrs = splitToChunks([...pages], 5);
+	let reports = {};
+
+	function reportGen() {
+		const promises = subArrs.map(async (pages, i) => {
+			for(let i = 0; i < pages.length; i += 1)
```

Comment:
Use forEach

Reply:
Can't. Some issues arise for the await statements with forEach.